README.md

```diff
@@ -5,13 +5,117 @@ Kubernetes Dashboard is a general purpose, web-based UI for Kubernetes clusters.
 manage applications running in the cluster, troubleshoot them, as well as, manage the cluster
 itself.
 
+## Usage
+
 The dashboard is currently under active development and is not ready for production use (yet!).
 
-# Contribute
+## Architecture Overview
+
+Kubernetes Dashboard project consists of two main components. They are called here the frontend and the backend. The frontend is a single page web application that runs in a browser. It fetches all its business data from the backend using standard HTTP methods. The backend implements UI-business logic and fetches raw data from the various Kubernetes APIs.
+
+## Preparation
+
+Make sure the following software is installed and added to the `$PATH` variable:
+* Docker (1.3+)
+* go (1.5+)
+* godep (18+)
+* nodejs (0.10.25+)
+* npm (1.3+)
+* etcd (2.2+)
+* java (7+)
+* gulp (3.9+)
+
+Clone the repository and install the dependencies:
+```
+$ npm install
+```
+
+## Run a Kubernetes Cluster
+
+For development it is recommended to run a local Kubernetes cluster. For your convenience, a task is provided that checks out the latest stable version, compiles and runs the script: `<kubernetes>/hack/local-up-cluster.sh`. Open a separate tab in your terminal and run the following command:
+
+```
+$ gulp local-up-cluster
+```
+Depending on your environment you might need to run the gulp task with sudo. Also, you might need to propagate the `$PATH` settings to the root user.
+```
+$ sudo env "PATH=$PATH" gulp local-up-cluster
+```
+This will build and start a lightweight local cluster, consisting of a master and a single node. All processes run locally, without virtual machines. The local cluster should behave like a real cluster, however, plugins like heapster are not installed. Type Control-C to shut it down.
+
+From time to time you might want to use to a real Kubernetes cluster (e.g. GCE, Vagrant) instead of the local one. The most convenient way is to create a proxy. Run the following command instead of the gulp task from above:
+```
+$ kubectl proxy --port=8080.
+```
+kubectl will handle authentication with Kubernetes and create an API proxy with the address `localhost:8080`. Therefore, no changes in the configuration are required.
+
+## Serving Dashboard for Development
+
+It is easy to compile and run Dashboard. Open a new tab in your terminal and type:
+```
+$ gulp serve
+```
+Open a browser and access the UI under `localhost:9090`. A lot of things happened underneath. Let's scratch on the surface a bit.
+
+Compilation:
+* Stylesheets are implemented with SASS and compiled to CSS with libsass
+* JavaScript is implemented in ES6. It is compiled with Babel for development and the Google-Closure-Compiler for production. This has to be done because currently browsers do not handle ES6 syntax and modules correctly.
```

Comment:
"This has to be done because currently browsers do not handle ES6 syntax and modules correctly." This is only partially true (for Babel & development). But in fact, we compile the code to make sure it is correct and save bytes on the wire when serving the UI to users.

I suggest to just remove this sentence. We do not need to justify technical decisions in developers manual.